% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\PassOptionsToPackage{dvipsnames,svgnames,x11names}{xcolor}
%
\documentclass[
  letterpaper,
  DIV=11,
  numbers=noendperiod]{scrartcl}

\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else  
    % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
% Make \paragraph and \subparagraph free-standing
\makeatletter
\ifx\paragraph\undefined\else
  \let\oldparagraph\paragraph
  \renewcommand{\paragraph}{
    \@ifstar
      \xxxParagraphStar
      \xxxParagraphNoStar
  }
  \newcommand{\xxxParagraphStar}[1]{\oldparagraph*{#1}\mbox{}}
  \newcommand{\xxxParagraphNoStar}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
  \let\oldsubparagraph\subparagraph
  \renewcommand{\subparagraph}{
    \@ifstar
      \xxxSubParagraphStar
      \xxxSubParagraphNoStar
  }
  \newcommand{\xxxSubParagraphStar}[1]{\oldsubparagraph*{#1}\mbox{}}
  \newcommand{\xxxSubParagraphNoStar}[1]{\oldsubparagraph{#1}\mbox{}}
\fi
\makeatother

\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{241,243,245}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.40,0.45,0.13}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\BuiltInTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.13,0.47,0.30}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textit{#1}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textit{#1}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\ExtensionTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.28,0.35,0.67}{#1}}
\newcommand{\ImportTok}[1]{\textcolor[rgb]{0.00,0.46,0.62}{#1}}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{#1}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.68,0.00,0.00}{#1}}
\newcommand{\RegionMarkerTok}[1]{\textcolor[rgb]{0.00,0.23,0.31}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.13,0.47,0.30}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.13,0.47,0.30}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.07,0.07,0.07}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.13,0.47,0.30}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.37,0.37,0.37}{\textit{#1}}}

\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother

\usepackage{listings}
\lstset{
  basicstyle=\ttfamily\small,  % Font style
  breaklines=true,             % Enable line wrapping
  breakatwhitespace=true,      % Break lines at whitespaces
  frame=single,                % Add a frame around code blocks
  prebreak=\raisebox{0ex}[0ex][0ex]{\ensuremath{\hookleftarrow}}, % Indicate broken lines
  postbreak=\raisebox{0ex}[0ex][0ex]{\ensuremath{\hookrightarrow}}
}
\KOMAoption{captions}{tableheading}
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\AtBeginDocument{%
\ifdefined\contentsname
  \renewcommand*\contentsname{Table of contents}
\else
  \newcommand\contentsname{Table of contents}
\fi
\ifdefined\listfigurename
  \renewcommand*\listfigurename{List of Figures}
\else
  \newcommand\listfigurename{List of Figures}
\fi
\ifdefined\listtablename
  \renewcommand*\listtablename{List of Tables}
\else
  \newcommand\listtablename{List of Tables}
\fi
\ifdefined\figurename
  \renewcommand*\figurename{Figure}
\else
  \newcommand\figurename{Figure}
\fi
\ifdefined\tablename
  \renewcommand*\tablename{Table}
\else
  \newcommand\tablename{Table}
\fi
}
\@ifpackageloaded{float}{}{\usepackage{float}}
\floatstyle{ruled}
\@ifundefined{c@chapter}{\newfloat{codelisting}{h}{lop}}{\newfloat{codelisting}{h}{lop}[chapter]}
\floatname{codelisting}{Listing}
\newcommand*\listoflistings{\listof{codelisting}{List of Listings}}
\makeatother
\makeatletter
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\@ifpackageloaded{subcaption}{}{\usepackage{subcaption}}
\makeatother

\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage{bookmark}

\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={Differential Gene Expression - Baggerley's Test},
  pdfauthor={Payel Sardar},
  colorlinks=true,
  linkcolor={blue},
  filecolor={Maroon},
  citecolor={Blue},
  urlcolor={Blue},
  pdfcreator={LaTeX via pandoc}}


\title{Differential Gene Expression - Baggerley's Test}
\author{Payel Sardar}
\date{}

\begin{document}
\maketitle


This script performs Differential Gene Expression Analysis using
Unpaired sample Baggerley's Test and creates corresponding
visualisations.

\section{Reading and Pre-processing
Datasets}\label{reading-and-pre-processing-datasets}

Setting the home working directory.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{setwd}\NormalTok{(}\StringTok{"D:/KCL2024/Courses/7BBG1002\_Cloud\_computing/Project"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Loading necessary packages

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(dplyr)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}

Attaching package: 'dplyr'
\end{verbatim}

\begin{verbatim}
The following objects are masked from 'package:stats':

    filter, lag
\end{verbatim}

\begin{verbatim}
The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(ggplot2)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
Warning: package 'ggplot2' was built under R version 4.4.2
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(tidyr)}
\FunctionTok{library}\NormalTok{(data.table)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
Warning: package 'data.table' was built under R version 4.4.2
\end{verbatim}

\begin{verbatim}

Attaching package: 'data.table'
\end{verbatim}

\begin{verbatim}
The following objects are masked from 'package:dplyr':

    between, first, last
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(ggplot2)}
\end{Highlighting}
\end{Shaded}

Reading the file on sample information

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{sample\_info }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"../metadata/sample\_types.csv"}\NormalTok{, }\AttributeTok{sep =} \StringTok{","}\NormalTok{, }\AttributeTok{header =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{head}\NormalTok{(sample\_info)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
         Run    BioSample  Bases   Bytes carbon_source Experiment GEO_Accession
1 SRR1166442 SAMN02639514 2.03 G 1.29 Gb       Glucose  SRX468698    GSM1324496
2 SRR1166443 SAMN02639516 2.25 G 1.42 Gb       Glucose  SRX468699    GSM1324497
3 SRR1166444 SAMN02639513 1.74 G 1.05 Gb       Glucose  SRX468700    GSM1324498
4 SRR1166445 SAMN02639515 2.28 G 1.44 Gb    Cellobiose  SRX468701    GSM1324499
5 SRR1166446 SAMN02639512 1.91 G 1.20 Gb    Cellobiose  SRX468702    GSM1324500
6 SRR1166447 SAMN02639517 1.73 G 1.04 Gb    Cellobiose  SRX468703    GSM1324501
          create_date Sample.Name            source_name
1 2014-02-1010:25:00Z  GSM1324496    Glucose-grown cells
2 2014-02-1010:25:00Z  GSM1324497    Glucose-grown cells
3 2014-02-1010:24:00Z  GSM1324498    Glucose-grown cells
4 2014-02-1010:25:00Z  GSM1324499 Cellobiose-grown cells
5 2014-02-1010:24:00Z  GSM1324500 Cellobiose-grown cells
6 2014-02-1010:24:00Z  GSM1324501 Cellobiose-grown cells
\end{verbatim}

Reading the Normalized RPKM data for Glucose and Cellobiose samples

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{glucose\_data }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"../data/processed/Normalized\_data/glucose\_merged.csv"}\NormalTok{, }\AttributeTok{header =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{cellobiose\_data }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"../data/processed/Normalized\_data/cellobiose\_merged.csv"}\NormalTok{, }\AttributeTok{header =} \ConstantTok{TRUE}\NormalTok{)}

\FunctionTok{head}\NormalTok{(glucose\_data)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
     Geneid SRR1166442 SRR1166443 SRR1166444
1   YAL068C  8.4379100 11.1595834  10.898455
2 YAL067W-A  0.3630822  0.3290228   0.564276
3   YAL067C 55.5291618 53.1827885  40.971760
4   YAL065C  3.4225421  3.5537864   2.825754
5 YAL064W-B  3.3315888  4.0691744   3.292351
6 YAL064C-A 10.1396180 10.1729361   8.273086
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(cellobiose\_data)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
     Geneid SRR1166445 SRR1166446 SRR1166447
1   YAL068C 11.6956377  9.3393236 10.1604814
2 YAL067W-A  0.5115573  0.4956395  0.1336905
3   YAL067C 15.5382521 17.9623442 15.2407221
4   YAL065C  1.8685722  2.3360375  2.5991929
5 YAL064W-B  3.0000621  2.7435795  2.8801365
6 YAL064C-A 10.1022500  7.8599846 10.0004739
\end{verbatim}

Setting Geneid as row names and dropping the Geneid column to obtain a
numeric dataset for downstream analysis

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gene\_ids }\OtherTok{\textless{}{-}}\NormalTok{ glucose\_data}\SpecialCharTok{$}\NormalTok{Geneid}

\FunctionTok{row.names}\NormalTok{(glucose\_data) }\OtherTok{\textless{}{-}}\NormalTok{ glucose\_data}\SpecialCharTok{$}\NormalTok{Geneid}
\NormalTok{glucose\_data }\OtherTok{\textless{}{-}}\NormalTok{ glucose\_data[, }\SpecialCharTok{{-}}\DecValTok{1}\NormalTok{]}

\FunctionTok{row.names}\NormalTok{(cellobiose\_data) }\OtherTok{\textless{}{-}}\NormalTok{ cellobiose\_data}\SpecialCharTok{$}\NormalTok{Geneid}
\NormalTok{cellobiose\_data }\OtherTok{\textless{}{-}}\NormalTok{ cellobiose\_data[, }\SpecialCharTok{{-}}\DecValTok{1}\NormalTok{]}
\end{Highlighting}
\end{Shaded}

\section{Differential Gene Expression
Analysis}\label{differential-gene-expression-analysis}

\subsection{Performing an unpaired sample analysis using the Baggerley's
test}\label{performing-an-unpaired-sample-analysis-using-the-baggerleys-test}

Function for performing Baggerley's test

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{baggerley\_test }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(prop1, prop2, }\AttributeTok{pseudo\_count =} \FloatTok{0.001}\NormalTok{) \{}
  
\NormalTok{  prop1 }\OtherTok{\textless{}{-}} \FunctionTok{as.numeric}\NormalTok{(prop1)}
\NormalTok{  prop2 }\OtherTok{\textless{}{-}} \FunctionTok{as.numeric}\NormalTok{(prop2)}
  
  \CommentTok{\# Adding pseudo{-}count to avoid zero mean}
\NormalTok{  prop1 }\OtherTok{\textless{}{-}}\NormalTok{ prop1 }\SpecialCharTok{+}\NormalTok{ pseudo\_count}
\NormalTok{  prop2 }\OtherTok{\textless{}{-}}\NormalTok{ prop2 }\SpecialCharTok{+}\NormalTok{ pseudo\_count}
  
  \CommentTok{\# Computing the sample size of each group}
\NormalTok{  n1 }\OtherTok{\textless{}{-}} \FunctionTok{length}\NormalTok{(prop1)}
\NormalTok{  n2 }\OtherTok{\textless{}{-}} \FunctionTok{length}\NormalTok{(prop2)}
  
  \CommentTok{\# Computing the mean for each group}
\NormalTok{  mean1 }\OtherTok{\textless{}{-}} \FunctionTok{mean}\NormalTok{(prop1)}
\NormalTok{  mean2 }\OtherTok{\textless{}{-}} \FunctionTok{mean}\NormalTok{(prop2)}
  
  \CommentTok{\# Compute the sample variance for each group}
\NormalTok{  var1 }\OtherTok{\textless{}{-}} \FunctionTok{var}\NormalTok{(prop1)  }\CommentTok{\# Variance for group 1}
\NormalTok{  var2 }\OtherTok{\textless{}{-}} \FunctionTok{var}\NormalTok{(prop2)  }\CommentTok{\# Variance for group 2}
  
  \CommentTok{\# Calculate pooled variance}
  \CommentTok{\#Adding 1e{-}6 prevent "Divide by zero" error in Z{-}score computation}
\NormalTok{  pooled\_var }\OtherTok{\textless{}{-}}\NormalTok{ (((n1 }\SpecialCharTok{{-}} \DecValTok{1}\NormalTok{) }\SpecialCharTok{*}\NormalTok{ var1) }\SpecialCharTok{+}\NormalTok{ ((n2 }\SpecialCharTok{{-}} \DecValTok{1}\NormalTok{) }\SpecialCharTok{*}\NormalTok{ var2)) }\SpecialCharTok{/}\NormalTok{ (n1 }\SpecialCharTok{+}\NormalTok{ n2 }\SpecialCharTok{{-}} \DecValTok{2}\NormalTok{) }\SpecialCharTok{+} \FloatTok{1e{-}6} 
  
  \CommentTok{\# Z{-}score using pooled standard deviation}
\NormalTok{  z }\OtherTok{\textless{}{-}}\NormalTok{ (mean1 }\SpecialCharTok{{-}}\NormalTok{ mean2) }\SpecialCharTok{/} \FunctionTok{sqrt}\NormalTok{(pooled\_var }\SpecialCharTok{*}\NormalTok{ (}\DecValTok{1}\SpecialCharTok{/}\NormalTok{n1 }\SpecialCharTok{+} \DecValTok{1}\SpecialCharTok{/}\NormalTok{n2))}
  
  \CommentTok{\# Two{-}tailed test}
\NormalTok{  p\_value }\OtherTok{\textless{}{-}} \DecValTok{2} \SpecialCharTok{*}\NormalTok{ (}\DecValTok{1} \SpecialCharTok{{-}} \FunctionTok{pnorm}\NormalTok{(}\FunctionTok{abs}\NormalTok{(z)))}
  
  \FunctionTok{return}\NormalTok{(p\_value)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

Iterating over rows(genes) to compute p-values for each gene using the
Baggerley's test.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{baggerley\_results }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\FunctionTok{nrow}\NormalTok{(glucose\_data), }\ControlFlowTok{function}\NormalTok{(i) \{}
\NormalTok{  prop1 }\OtherTok{\textless{}{-}}\NormalTok{ cellobiose\_data[i, ] }
\NormalTok{  prop2 }\OtherTok{\textless{}{-}}\NormalTok{ glucose\_data[i, ]   }
  \FunctionTok{baggerley\_test}\NormalTok{(prop1, prop2, }\FloatTok{0.001}\NormalTok{)}
\NormalTok{\})}
\end{Highlighting}
\end{Shaded}

Creating a dataframe to store the results from the Baggerley's test

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Storing results (p{-}values) in a data frame}
\NormalTok{result\_df2 }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \CommentTok{\# Mean Normalised RPKM for glucose}
  \AttributeTok{glucose\_norm\_rpkm\_mean =} \FunctionTok{rowMeans}\NormalTok{(glucose\_data),}
  \CommentTok{\# Mean Normalised RPKM for cellobiose samples}
  \AttributeTok{cellobiose\_norm\_rpkm\_mean =} \FunctionTok{rowMeans}\NormalTok{(cellobiose\_data),}
  \CommentTok{\# P{-}values from Baggerley\textquotesingle{}s test}
  \AttributeTok{p\_value =}\NormalTok{ baggerley\_results   }
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

FDR correction for P-value

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{result\_df2}\SpecialCharTok{$}\NormalTok{adjusted\_p\_value }\OtherTok{\textless{}{-}} \FunctionTok{p.adjust}\NormalTok{(result\_df2}\SpecialCharTok{$}\NormalTok{p\_value, }\AttributeTok{method =} \StringTok{"fdr"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Computing the fold change and log2 transformed fold change

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Pseudo{-}count of 0.001 is added to prevent 0/0 error or log(0)}
\NormalTok{result\_df2}\SpecialCharTok{$}\NormalTok{fold\_change }\OtherTok{\textless{}{-}}\NormalTok{ (result\_df2}\SpecialCharTok{$}\NormalTok{cellobiose\_norm\_rpkm\_mean }\SpecialCharTok{+} \FloatTok{0.001}\NormalTok{) }\SpecialCharTok{/}
\NormalTok{  (result\_df2}\SpecialCharTok{$}\NormalTok{glucose\_norm\_rpkm\_mean }\SpecialCharTok{+} \FloatTok{0.001}\NormalTok{)}
\NormalTok{result\_df2}\SpecialCharTok{$}\NormalTok{log2\_fold\_change }\OtherTok{\textless{}{-}}  \FunctionTok{log2}\NormalTok{(}\FunctionTok{abs}\NormalTok{(result\_df2}\SpecialCharTok{$}\NormalTok{fold\_change))}
\end{Highlighting}
\end{Shaded}

Extracting the Differentially Expressed genes based on the significance
threshold used in the original study

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{significant\_genes }\OtherTok{\textless{}{-}}\NormalTok{ result\_df2[result\_df2}\SpecialCharTok{$}\NormalTok{adjusted\_p\_value }\SpecialCharTok{\textless{}=} \FloatTok{0.001} \SpecialCharTok{\&} 
                                  \FunctionTok{abs}\NormalTok{(result\_df2}\SpecialCharTok{$}\NormalTok{log2\_fold\_change) }\SpecialCharTok{\textgreater{}=} \FloatTok{1.0}\NormalTok{, ]}
\FunctionTok{head}\NormalTok{(significant\_genes)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
          glucose_norm_rpkm_mean cellobiose_norm_rpkm_mean      p_value
YAL067C                49.894570                16.2471061 2.406964e-13
YAL062W               141.841053               331.2851169 0.000000e+00
YAL054C                18.816943                53.6567692 0.000000e+00
YAL044C              1128.504614               423.9457399 0.000000e+00
YAL039C               110.846294               388.6617995 0.000000e+00
YNCA0001W               2.994296                 0.9976067 2.933395e-07
          adjusted_p_value fold_change log2_fold_change
YAL067C       6.342515e-13   0.3256423        -1.618640
YAL062W       0.000000e+00   2.3355987         1.223792
YAL054C       0.000000e+00   2.8514152         1.511678
YAL044C       0.000000e+00   0.3756709        -1.412459
YAL039C       0.000000e+00   3.5062904         1.809945
YNCA0001W     5.804949e-07   0.3333916        -1.584710
\end{verbatim}

\subsection{Annotating the Differentially
Genes}\label{annotating-the-differentially-genes}

The GAF file used in the analysis was modified by removing the header
(comment section) that begins with `!'.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gaf\_data }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{fread}\NormalTok{(}\StringTok{"../metadata/sgd.gaf/sgd\_noheader.gaf"}\NormalTok{, }
                                \AttributeTok{sep =} \StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\AttributeTok{quote =} \StringTok{""}\NormalTok{, }\AttributeTok{fill =} \ConstantTok{TRUE}\NormalTok{))}
\NormalTok{gaf\_data }\OtherTok{\textless{}{-}}\NormalTok{ gaf\_data[gaf\_data[,}\DecValTok{1}\NormalTok{]}\SpecialCharTok{==}\StringTok{"SGD"}\NormalTok{,]}
\NormalTok{gaf\_data }\OtherTok{\textless{}{-}}\NormalTok{ gaf\_data[, }\FunctionTok{c}\NormalTok{(}\DecValTok{3}\NormalTok{, }\DecValTok{10}\NormalTok{,}\DecValTok{11}\NormalTok{)]}
\FunctionTok{colnames}\NormalTok{(gaf\_data) }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"feature\_id"}\NormalTok{, }\StringTok{"description"}\NormalTok{, }\StringTok{"synonym"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Cleaning up the gene synonyms and preparing to merge with DEG results

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gaf\_data}\SpecialCharTok{$}\NormalTok{gene\_id }\OtherTok{\textless{}{-}} \FunctionTok{sub}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{|.*"}\NormalTok{, }\StringTok{""}\NormalTok{, gaf\_data}\SpecialCharTok{$}\NormalTok{synonym)}
\NormalTok{gaf\_data }\OtherTok{\textless{}{-}}\NormalTok{ gaf\_data }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

Annotating the significant DEGs with the additional information from the
GAF file

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{significant\_genes}\SpecialCharTok{$}\NormalTok{gene\_id }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(significant\_genes)}
\NormalTok{annotated\_degs }\OtherTok{\textless{}{-}}\NormalTok{ gaf\_data }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{right\_join}\NormalTok{(significant\_genes, }\AttributeTok{by =} \StringTok{"gene\_id"}\NormalTok{)}
\NormalTok{annotated\_degs}\SpecialCharTok{$}\NormalTok{feature\_id }\OtherTok{\textless{}{-}} \FunctionTok{coalesce}\NormalTok{(annotated\_degs}\SpecialCharTok{$}\NormalTok{feature\_id, annotated\_degs}\SpecialCharTok{$}\NormalTok{gene\_id)}
\FunctionTok{head}\NormalTok{(annotated\_degs)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
  feature_id
1       ATM1
2    YJL218W
3    YPL264C
4       SAM1
5       CYC1
6       TMT1
                                                          description
1 Mitochondrial inner membrane ATP-binding cassette (ABC) transporter
2                   Mitochondrial protein, putative acetyltransferase
3                   Endoplasmic reticulum protein of unknown function
4                                     S-adenosylmethionine synthetase
5                                             Cytochrome c, isoform 1
6                                   Trans-aconitate methyltransferase
                                                               synonym gene_id
1 YMR301C|ATP-binding cassette Fe/S cluster precursor transporter ATM1 YMR301C
2                                            YJL218W|acetyltransferase YJL218W
3                                                              YPL264C YPL264C
4                    YLR180W|ETH10|methionine adenosyltransferase SAM1 YLR180W
5                                       YJR048W|cytochrome c isoform 1 YJR048W
6                     YER175C|TAM1|trans-aconitate 3-methyltransferase YER175C
  glucose_norm_rpkm_mean cellobiose_norm_rpkm_mean     p_value adjusted_p_value
1              54.439460                  21.50717 0.000000000     0.0000000000
2              27.132087                  12.11236 0.000000000     0.0000000000
3              73.559829                  17.60086 0.000000000     0.0000000000
4            1224.673346                 167.15200 0.000000000     0.0000000000
5               9.374713                  24.66442 0.000152921     0.0002562663
6             361.766324                  81.70760 0.000000000     0.0000000000
  fold_change log2_fold_change
1   0.3950769        -1.339794
2   0.4464426        -1.163454
3   0.2392830        -2.063210
4   0.1364877        -2.873157
5   2.6307783         1.395490
6   0.2258595        -2.146502
\end{verbatim}

Exporting the results for downstream analysis

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{write.csv}\NormalTok{(annotated\_degs, }\StringTok{"../Output/results/Annotated\_degs\_Baggerley.csv"}\NormalTok{, }
          \AttributeTok{row.names =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{fileEncoding =} \StringTok{"UTF{-}8"}\NormalTok{)}
\FunctionTok{write}\NormalTok{(annotated\_degs}\SpecialCharTok{$}\NormalTok{feature\_id, }\StringTok{"../Output/results/FeatureId.txt"}\NormalTok{, }\AttributeTok{ncolumns =} \DecValTok{1}\NormalTok{)}
\FunctionTok{write}\NormalTok{(annotated\_degs}\SpecialCharTok{$}\NormalTok{gene\_id, }\StringTok{"../Output/results/GeneId.txt"}\NormalTok{, }\AttributeTok{ncolumns =} \DecValTok{1}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Extracting information about the transcription factors
mentioned in the original
study}\label{extracting-information-about-the-transcription-factors-mentioned-in-the-original-study}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{tf\_list }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{\textquotesingle{}MET32\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}MET28\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}THI2\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}MIG2\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}UGA3\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}SIP4\textquotesingle{}}\NormalTok{, }
             \StringTok{\textquotesingle{}MIG3\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}HMS2\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}KAR4\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}MAL13\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}YAP5\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}DAL80\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}ADR1\textquotesingle{}}\NormalTok{, }
             \StringTok{\textquotesingle{}USV1\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}CAT8\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}GSM1\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}XBP1\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}SUT1\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}HAP4\textquotesingle{}}\NormalTok{)}
\NormalTok{tf\_dge }\OtherTok{\textless{}{-}}\NormalTok{ annotated\_degs }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(feature\_id }\SpecialCharTok{\%in\%}\NormalTok{ tf\_list)}

\CommentTok{\#write.csv(tf\_dge, "../Output/results/TranscriptionFactors\_degs.csv", }
\CommentTok{\#          row.names = FALSE, fileEncoding = "UTF{-}8")}
\end{Highlighting}
\end{Shaded}

\subsection{Data Visualisation}\label{data-visualisation}

\subsubsection{Volcano plot for all differentially expressed
genes}\label{volcano-plot-for-all-differentially-expressed-genes}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Creating a new column in the data frame to indicate significance}
\NormalTok{result\_df2}\SpecialCharTok{$}\NormalTok{Significance }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(result\_df2}\SpecialCharTok{$}\NormalTok{adjusted\_p\_value }\SpecialCharTok{\textless{}=} \FloatTok{0.001} \SpecialCharTok{\&} 
                                    \FunctionTok{abs}\NormalTok{(result\_df2}\SpecialCharTok{$}\NormalTok{log2\_fold\_change) }\SpecialCharTok{\textgreater{}=} \FloatTok{1.0}\NormalTok{, }
                                   \StringTok{"Significant"}\NormalTok{, }\StringTok{"Non{-}significant"}\NormalTok{)}

\CommentTok{\# Create the volcano plot with the updated \textquotesingle{}Significance\textquotesingle{} column}
\NormalTok{volcano\_plot }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(result\_df2, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ log2\_fold\_change,}
                                       \AttributeTok{y =} \SpecialCharTok{{-}}\FunctionTok{log10}\NormalTok{(adjusted\_p\_value))) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{size =} \FloatTok{0.7}\NormalTok{, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{color =}\NormalTok{ Significance)) }\SpecialCharTok{+} 
  \FunctionTok{geom\_hline}\NormalTok{(}\AttributeTok{yintercept =} \SpecialCharTok{{-}}\FunctionTok{log10}\NormalTok{(}\FloatTok{0.001}\NormalTok{), }\AttributeTok{color =} \StringTok{"red"}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_vline}\NormalTok{(}\AttributeTok{xintercept =} \DecValTok{0}\NormalTok{, }\AttributeTok{color =} \StringTok{"blue"}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{) }\SpecialCharTok{+} 
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{x =} \StringTok{"Log2 Fold Change"}\NormalTok{, }\AttributeTok{y =} \StringTok{"{-}Log10(Adjusted P{-}Value)"}\NormalTok{, }
       \AttributeTok{color =} \StringTok{"Significance"}\NormalTok{) }\SpecialCharTok{+}
  
  \CommentTok{\# Custom color scale for significance}
  \FunctionTok{scale\_color\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"Non{-}significant"} \OtherTok{=} \StringTok{"red"}\NormalTok{, }\StringTok{"Significant"} \OtherTok{=} \StringTok{"blue"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{theme\_minimal}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{legend.position =} \StringTok{"top"}\NormalTok{, }
    \AttributeTok{legend.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{12}\NormalTok{),  }
    \AttributeTok{legend.text =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{10}\NormalTok{),    }
    \AttributeTok{panel.grid.minor =} \FunctionTok{element\_blank}\NormalTok{()   }
\NormalTok{  )}

\FunctionTok{print}\NormalTok{(volcano\_plot)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Differential_Gene_Expression_files/figure-pdf/unnamed-chunk-17-1.pdf}

Saving the plot as jpg for preparing reports

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}
  \AttributeTok{filename =} \StringTok{"../Output/plots/VolcanoPlot\_Baggerley.jpg"}\NormalTok{,   }
  \AttributeTok{plot =}\NormalTok{ volcano\_plot,                      }
  \AttributeTok{width =} \DecValTok{6}\NormalTok{, }\AttributeTok{height =} \DecValTok{4}\NormalTok{,               }
  \AttributeTok{dpi =} \DecValTok{300}                            
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Visualising the top 10 upregulated and downregulated
differentially expressed
genes}\label{visualising-the-top-10-upregulated-and-downregulated-differentially-expressed-genes}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# sorting the annotated DEGs based on their log2(fold change) in ascending order}
\NormalTok{deg\_sorted }\OtherTok{\textless{}{-}}\NormalTok{ annotated\_degs[}\FunctionTok{order}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{annotated\_degs}\SpecialCharTok{$}\NormalTok{log2\_fold\_change), ]}
\NormalTok{top\_upregulated }\OtherTok{\textless{}{-}} \FunctionTok{head}\NormalTok{(deg\_sorted, }\DecValTok{10}\NormalTok{)}
\NormalTok{top\_downregulated }\OtherTok{\textless{}{-}} \FunctionTok{tail}\NormalTok{(deg\_sorted, }\DecValTok{10}\NormalTok{)}
\NormalTok{top\_genes }\OtherTok{\textless{}{-}} \FunctionTok{rbind}\NormalTok{(top\_upregulated, top\_downregulated)}

\CommentTok{\# Making a bar}
\NormalTok{top10 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(top\_genes, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =} \FunctionTok{reorder}\NormalTok{(feature\_id, log2\_fold\_change), }
                      \AttributeTok{y =}\NormalTok{ log2\_fold\_change, }
                      \AttributeTok{fill =}\NormalTok{ log2\_fold\_change }\SpecialCharTok{\textless{}} \DecValTok{0}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{stat =} \StringTok{"identity"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"skyblue"}\NormalTok{, }\StringTok{"pink"}\NormalTok{),}
                    \AttributeTok{labels =} \FunctionTok{c}\NormalTok{(}\StringTok{"Top Upregulated"}\NormalTok{, }\StringTok{"Top Downregulated"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{x =} \StringTok{"Genes"}\NormalTok{, }
       \AttributeTok{y =} \StringTok{"Log2 Fold Change"}\NormalTok{,}
       \AttributeTok{fill =} \StringTok{"Regulation Status"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme\_minimal}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{panel.grid.major =} \FunctionTok{element\_blank}\NormalTok{(), }\CommentTok{\# Remove major gridlines}
    \AttributeTok{panel.grid.minor =} \FunctionTok{element\_blank}\NormalTok{(), }\CommentTok{\# Remove minor gridlines}
    \AttributeTok{legend.position =} \StringTok{"top"} \CommentTok{\# Move legend to the top}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{coord\_flip}\NormalTok{()  }\CommentTok{\# Flip for better readability}
\FunctionTok{print}\NormalTok{(top10)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Differential_Gene_Expression_files/figure-pdf/unnamed-chunk-19-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}
  \AttributeTok{filename =} \StringTok{"../Output/plots/Top\_DEG\_Baggerley.jpg"}\NormalTok{,   }
  \AttributeTok{plot =}\NormalTok{ top10,                      }
  \AttributeTok{width =} \DecValTok{6}\NormalTok{, }\AttributeTok{height =} \DecValTok{4}\NormalTok{,               }
  \AttributeTok{dpi =} \DecValTok{300}                            
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Preparing a bar plot to compare the fold change of
differentially expressed transcription
factors.}\label{preparing-a-bar-plot-to-compare-the-fold-change-of-differentially-expressed-transcription-factors.}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{tf\_fc\_plot }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(tf\_dge, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =} \FunctionTok{reorder}\NormalTok{(feature\_id, fold\_change), }
                   \AttributeTok{y =}\NormalTok{ log2\_fold\_change, }
                   \AttributeTok{fill =}\NormalTok{ log2\_fold\_change }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{stat =} \StringTok{"identity"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"pink"}\NormalTok{, }\StringTok{"skyblue"}\NormalTok{), }
                    \AttributeTok{labels =} \FunctionTok{c}\NormalTok{(}\StringTok{"Downregulated"}\NormalTok{, }\StringTok{"Upregulated"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{x =} \StringTok{"Transcription Factors"}\NormalTok{, }
    \AttributeTok{y =} \StringTok{"Log2 transformed Fold Change (C8 vs G8)"}\NormalTok{, }
    \AttributeTok{fill =} \StringTok{"Expression Levels"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_minimal}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{panel.grid.major =} \FunctionTok{element\_blank}\NormalTok{(), }\CommentTok{\# Remove major gridlines}
    \AttributeTok{panel.grid.minor =} \FunctionTok{element\_blank}\NormalTok{(), }\CommentTok{\# Remove minor gridlines}
    \AttributeTok{legend.position =} \StringTok{"top"} \CommentTok{\# Move legend to the top}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{coord\_flip}\NormalTok{()  }\CommentTok{\# Flip for better readability}
\FunctionTok{print}\NormalTok{(tf\_fc\_plot)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Differential_Gene_Expression_files/figure-pdf/unnamed-chunk-21-1.pdf}

Saving the barplot

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}
  \AttributeTok{filename =} \StringTok{"../Output/plots/Fold\_change\_tf\_Baggerley.jpg"}\NormalTok{,   }
  \AttributeTok{plot =}\NormalTok{ tf\_fc\_plot,                      }
  \AttributeTok{width =} \DecValTok{6}\NormalTok{, }\AttributeTok{height =} \DecValTok{4}\NormalTok{,               }
  \AttributeTok{dpi =} \DecValTok{300}                            
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\section{Comparing with the DeSeq
results}\label{comparing-with-the-deseq-results}

Reading the DeSEQ results file

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{deseq }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"../Output/results/sig\_DEG\_DeSEQ.csv"}\NormalTok{, }\AttributeTok{header=}\ConstantTok{TRUE}\NormalTok{)}
\NormalTok{deseq\_annotated }\OtherTok{\textless{}{-}}\NormalTok{ gaf\_data }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{right\_join}\NormalTok{(deseq, }\AttributeTok{by =} \StringTok{"gene\_id"}\NormalTok{)}
\NormalTok{deseq\_annotated}\SpecialCharTok{$}\NormalTok{log2FoldChange }\OtherTok{\textless{}{-}}\NormalTok{ (}\SpecialCharTok{{-}}\DecValTok{1}\NormalTok{)}\SpecialCharTok{*}\NormalTok{deseq\_annotated}\SpecialCharTok{$}\NormalTok{log2FoldChange}
\end{Highlighting}
\end{Shaded}

Making the Bar plot for the Top upregulated and Downregulated genes

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# sorting the annotated DEGs based on their log2(fold change) in ascending order}
\NormalTok{deseq\_sorted }\OtherTok{\textless{}{-}}\NormalTok{ deseq\_annotated[}\FunctionTok{order}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{deseq\_annotated}\SpecialCharTok{$}\NormalTok{log2FoldChange), ]}
\NormalTok{top\_upregulated }\OtherTok{\textless{}{-}} \FunctionTok{head}\NormalTok{(deseq\_sorted, }\DecValTok{10}\NormalTok{)}
\NormalTok{top\_downregulated }\OtherTok{\textless{}{-}} \FunctionTok{tail}\NormalTok{(deseq\_sorted, }\DecValTok{10}\NormalTok{)}
\NormalTok{top\_genes }\OtherTok{\textless{}{-}} \FunctionTok{rbind}\NormalTok{(top\_upregulated, top\_downregulated)}

\CommentTok{\# Making a bar}
\NormalTok{top10 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(top\_genes, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =} \FunctionTok{reorder}\NormalTok{(feature\_id, log2FoldChange), }
                      \AttributeTok{y =}\NormalTok{ log2FoldChange, }
                      \AttributeTok{fill =}\NormalTok{ log2FoldChange }\SpecialCharTok{\textless{}} \DecValTok{0}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{stat =} \StringTok{"identity"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"skyblue"}\NormalTok{, }\StringTok{"pink"}\NormalTok{),}
                    \AttributeTok{labels =} \FunctionTok{c}\NormalTok{(}\StringTok{"Top Upregulated"}\NormalTok{, }\StringTok{"Top Downregulated"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{x =} \StringTok{"Genes"}\NormalTok{, }
       \AttributeTok{y =} \StringTok{"Log2 Fold Change"}\NormalTok{,}
       \AttributeTok{fill =} \StringTok{"Regulation Status"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme\_minimal}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{panel.grid.major =} \FunctionTok{element\_blank}\NormalTok{(), }\CommentTok{\# Remove major gridlines}
    \AttributeTok{panel.grid.minor =} \FunctionTok{element\_blank}\NormalTok{(), }\CommentTok{\# Remove minor gridlines}
    \AttributeTok{legend.position =} \StringTok{"top"} \CommentTok{\# Move legend to the top}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{coord\_flip}\NormalTok{()  }\CommentTok{\# Flip for better readability}
\FunctionTok{print}\NormalTok{(top10)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Differential_Gene_Expression_files/figure-pdf/unnamed-chunk-24-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}
  \AttributeTok{filename =} \StringTok{"../Output/plots/Top\_DEG\_DESeq.jpg"}\NormalTok{,   }
  \AttributeTok{plot =}\NormalTok{ top10,                      }
  \AttributeTok{width =} \DecValTok{6}\NormalTok{, }\AttributeTok{height =} \DecValTok{4}\NormalTok{,               }
  \AttributeTok{dpi =} \DecValTok{300}                            
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Visualising the Expression profile of the differentially Transcription
factors

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{tf\_deseq }\OtherTok{\textless{}{-}}\NormalTok{ deseq\_annotated }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(feature\_id }\SpecialCharTok{\%in\%}\NormalTok{ tf\_list)}

\NormalTok{tf\_fc\_plot }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(tf\_deseq, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =} \FunctionTok{reorder}\NormalTok{(feature\_id, log2FoldChange), }
                   \AttributeTok{y =}\NormalTok{ log2FoldChange, }
                   \AttributeTok{fill =}\NormalTok{ log2FoldChange }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{stat =} \StringTok{"identity"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"pink"}\NormalTok{, }\StringTok{"skyblue"}\NormalTok{), }
                    \AttributeTok{labels =} \FunctionTok{c}\NormalTok{(}\StringTok{"Downregulated"}\NormalTok{, }\StringTok{"Upregulated"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{x =} \StringTok{"Transcription Factors"}\NormalTok{, }
    \AttributeTok{y =} \StringTok{"Log2 transformed Fold Change (C8 vs G8)"}\NormalTok{, }
    \AttributeTok{fill =} \StringTok{"Expression Levels"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_minimal}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{panel.grid.major =} \FunctionTok{element\_blank}\NormalTok{(), }\CommentTok{\# Remove major gridlines}
    \AttributeTok{panel.grid.minor =} \FunctionTok{element\_blank}\NormalTok{(), }\CommentTok{\# Remove minor gridlines}
    \AttributeTok{legend.position =} \StringTok{"top"} \CommentTok{\# Move legend to the top}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{coord\_flip}\NormalTok{()  }\CommentTok{\# Flip for better readability}
\FunctionTok{print}\NormalTok{(tf\_fc\_plot)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Differential_Gene_Expression_files/figure-pdf/unnamed-chunk-26-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}
  \AttributeTok{filename =} \StringTok{"../Output/plots/Fold\_change\_tf\_DESeq.jpg"}\NormalTok{,   }
  \AttributeTok{plot =}\NormalTok{ tf\_fc\_plot,                      }
  \AttributeTok{width =} \DecValTok{6}\NormalTok{, }\AttributeTok{height =} \DecValTok{4}\NormalTok{,               }
  \AttributeTok{dpi =} \DecValTok{300}                            
\NormalTok{)}
\end{Highlighting}
\end{Shaded}





\end{document}
